### PR TITLE
SuperPMI collections: exclude more files/directories

### PR DESCRIPTION
**Pull request by BruceForstall**
Exclude a few more native code files and the 'runtimes' directory that contains uninteresting files, including native code files, to reduce the number of PMI exceptions during collections.

### CODE REVIEW

File: src/coreclr/scripts/superpmi_collect_setup.py [hunk 1]
```diff
@@ -559,7 +567,11 @@ def make_readable(folder_name):
         #     copy_directory(dotnet_src_directory, dotnet_dst_directory, verbose_output=False)
 
         input_artifacts = os.path.join(workitem_payload_directory, "collectAssembliesDirectory", coreclr_args.collection_name)
-        exclude_directory = ['Core_Root'] if coreclr_args.collection_name == "coreclr_tests" else []
+
+        exclude_directories = directories_to_ignore
+        if coreclr_args.collection_name == "coreclr_tests":
+            exclude_directories += ['Core_Root']
```
Comment (jakobbotsch) [inline src/coreclr/scripts/superpmi_collect_setup.py:573]: I think this might mutate `directories_to_ignore` too. Not sure if that is intentional.

Comment (BruceForstall) [reply src/coreclr/scripts/superpmi_collect_setup.py:573]: Well, not intentional, but also not problematic. `exclude_directories = directories_to_ignore` needs to be a copy; what's the better python-ic way?

Comment (jakobbotsch) [reply src/coreclr/scripts/superpmi_collect_setup.py:573]: I think `exclude_directories = list(directories_to_ignore)` would work.

Comment (BruceForstall) [reply src/coreclr/scripts/superpmi_collect_setup.py]: ok, I updated the code to use `list()`

**Review by jakobbotsch — APPROVED** [review]

**Pull request merged**